test/spec/modules/driftpixelBidAdapter_spec.js

```diff
@@ -0,0 +1,455 @@
+      const bidderRequest = {
+        gdprConsent: {
+          gdprApplies: true,
+          consentString: 'qwerty'
+        }
+      };
+      const request = JSON.parse(spec.buildRequests([defaultRequest], bidderRequest).data)[0];
+      expect(request).to.have.property('gdprApplies').and.equals(1);
+      expect(request).to.have.property('consentString').and.equals('qwerty');
+    });
+
+    it('should build request with usp consent data if applies', function () {
+      const bidderRequest = {
+        uspConsent: '1YA-'
+      };
+      const request = JSON.parse(spec.buildRequests([defaultRequest], bidderRequest).data)[0];
+      expect(request).to.have.property('usPrivacy').and.equals('1YA-');
+    });
+
+    it('should build request with coppa 1', function () {
+      config.setConfig({
+        coppa: true
+      });
+      const request = JSON.parse(spec.buildRequests([defaultRequest], {}).data)[0];
+      expect(request).to.have.property('coppa').and.equals(1);
+    });
+
+    it('should build request with extended ids', function () {
+      const idRequest = deepClone(defaultRequest);
+      idRequest.userIdAsEids = [
+        {source: 'adserver.org', uids: [{id: 'TTD_ID_FROM_USER_ID_MODULE', atype: 1, ext: {rtiPartner: 'TDID'}}]},
+        {source: 'pubcid.org', uids: [{id: 'pubCommonId_FROM_USER_ID_MODULE', atype: 1}]}
+      ];
+      const request = JSON.parse(spec.buildRequests([idRequest], {}).data)[0];
+      expect(request).to.have.property('userEids').and.deep.equal(idRequest.userIdAsEids);
+    });
+
+    it('should build request with video', function () {
+      const request = JSON.parse(spec.buildRequests([defaultRequestVideo], {}).data)[0];
+      expect(request).to.have.property('video').and.to.deep.equal({
+        playerSize: [640, 480],
+        context: 'instream',
+        skipppable: true
+      });
+      expect(request).to.have.property('sizes').and.to.deep.equal(['640x480']);
+    });
+  });
+
+  describe('interpretResponse', function () {
+    it('should return empty bids', function () {
+      const serverResponse = {
+        body: {
+          data: null
+        }
+      };
+
+      const invalidResponse = spec.interpretResponse(serverResponse, {});
+      expect(invalidResponse).to.be.an('array').that.is.empty;
+    });
+
+    it('should interpret valid response', function () {
+      const serverResponse = {
+        body: {
+          data: [{
+            requestId: 'qwerty',
+            cpm: 1,
+            currency: 'USD',
+            width: 300,
+            height: 250,
+            ttl: 600,
+            meta: {
+              advertiserDomains: ['driftpixel']
+            },
+            ext: {
+              pixels: [
+                ['iframe', 'surl1'],
+                ['image', 'surl2'],
+              ]
+            }
+          }]
+        }
+      };
+
+      const validResponse = spec.interpretResponse(serverResponse, {bidderRequest: defaultRequest});
```

Comment:
I think the format of the bidderRequest is not what will be present with a real bid (this is linked to the first comment)